src/types/targetValidators.spec.ts

```diff
@@ -602,3 +604,25 @@ describe('validateAuthConfig', () => {
     })
   })
 })
+
+describe('validateDocConfig', () => {
+  it('should throw an error when docConfig is null', () => {
+    expect(() =>
+      validateDocConfig((null as unknown) as DocConfig)
+    ).toThrowError('Invalid doc config: JSON cannot be null or undefined.')
+  })
```

Comment:
None of the attributes are mandatory - so am not sure this test is valid?

If the dataControllerUrl is not provided, then we simply shouldn't link to anything (for the tables)

displayMacroCore defaults to true

and outDirectory defaults to sasjsbuild/docs

Reply:
It's checking to have json object if provided.

++ adding default values
```
  if (docConfig.displayMacroCore === undefined) {
    docConfig.displayMacroCore = true
  }

  if (!docConfig.outDirectory) {
    docConfig.outDirectory = ''
  }

  if (!docConfig.dataControllerUrl) {
    docConfig.dataControllerUrl = ''
  }
```

In Cli code, we are using default behavior if "" empty string is present or property is not present

Reply:
aha - ok, great!